contracts/chain-adapters/Mode_Adapter.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "./interfaces/AdapterInterface.sol";
+import "../external/interfaces/WETH9Interface.sol";
+
+// @dev Use local modified CrossDomainEnabled contract instead of one exported by eth-optimism because we need
+// this contract's state variables to be `immutable` because of the delegateCall call.
+import "./CrossDomainEnabled.sol";
+import "@eth-optimism/contracts/L1/messaging/IL1StandardBridge.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../libraries/CircleCCTPAdapter.sol";
+import "../external/interfaces/CCTPInterfaces.sol";
+
+/**
+ * @notice Contract containing logic to send messages from L1 to Mode. This is a clone of the Mode adapter
+ * @dev Public functions calling external contracts do not guard against reentrancy because they are expected to be
+ * called via delegatecall, which will execute this contract's logic within the context of the originating contract.
+ * For example, the HubPool will delegatecall these functions, therefore its only necessary that the HubPool's methods
+ * that call this contract's logic guard against reentrancy.
+ */
+
+// solhint-disable-next-line contract-name-camelcase
+contract Mode_Adapter is CrossDomainEnabled, AdapterInterface, CircleCCTPAdapter {
+    using SafeERC20 for IERC20;
+    uint32 public constant L2_GAS_LIMIT = 200_000;
+
+    WETH9Interface public immutable L1_WETH;
+
+    IL1StandardBridge public immutable L1_STANDARD_BRIDGE;
+
+    /**
+     * @notice Constructs new Adapter.
+     * @param _l1Weth WETH address on L1.
+     * @param _crossDomainMessenger XDomainMessenger Mode system contract.
+     * @param _l1StandardBridge Standard bridge contract.
+     * @param _l1Usdc USDC address on L1.
+     * @param _cctpTokenMessenger TokenMessenger contract to bridge via CCTP.
+     */
+    constructor(
+        WETH9Interface _l1Weth,
+        address _crossDomainMessenger,
+        IL1StandardBridge _l1StandardBridge,
+        IERC20 _l1Usdc,
+        ITokenMessenger _cctpTokenMessenger
+    )
+        CrossDomainEnabled(_crossDomainMessenger)
+        CircleCCTPAdapter(_l1Usdc, _cctpTokenMessenger, CircleDomainIds.Ethereum) // CCTP is not enabled on this chain so Ethereum is a placeholder
```

Comment:
@nicholaspai @mrice32 we're setting this to Ethereum as a placeholder even though it's a no-op since `_cctpTokenMessager` will be zero.

We *need* to ensure that `_cctpTokenMessenger` is empty because this domain will incorrectly route + the `ZERO_ADDRESS` we need to pass for `_l1Usdc` is also invalid.